nservicebus/upgrades/5to6/index.md

```diff
@@ -10,20 +10,189 @@ related:
+
+**Do not upgrade an endpoint unless there is a compelling reason to do so.**
+
+Note that some new features added in Version 6 require that all endpoints are running on Version 6 before they can be switched on (can we identify even a subset of these?). Another factor to consider is the investment required to maintain codebases using different versions of NServiceBus. It may be cheaper in the long run to maintain a single codebase containing just Version 6 code than to invest in training and knowledge around Versions 5 and below.
+
+Once the list of endpoints that need to be upgraded to Version 6 has been identified, upgrade them one at a time. As a Version 6 endpoint is able to exchange messages with Version 5 endpoints, upgrade one endpoint, test it, and deploy it to production before upgrading the next endpoint. This keeps the scope of changes to a minimum which help to reduce risk and to isolate potential problems when they arise. 
+
+**Upgrade one endpoint at a time.**
+
+There is one common issue with upgrading a single endpoint at a time. If the endpoints in a solution share a common libary then upgrading one endpoint might lead to changes in the common library and that necessitates changes in all of the other endpoints that rely on the common library at the same time. The recommended approach to dealing with this is to create a copy of the common libary for the new endpoint and to upgrade it along with the endpoint. When the time comes to upgrade the second endpoint, change it's dependency to point to the new, upgraded, version of the common library. When using this approach, other changes to the common library should be minimized as they will need to be reflected in both codebases.
+
+The process of upgrading each endpoint is going to follow a common sequence of steps. Being able to repeatably apply those steps is key to the success of the upgrade project. The recommended approach is to upgrade a simple and low risk endpoint first to ensure that the process is well understood before tackling the endpoints that make up the core of the solution. Endpoints that send email or generate documents are often good candidates for this. When selecting the first endpoint to upgrade look for a small number of reasonably straightforward handlers and a small amount of NServiceBus configuration. It is worth considering selecting a simple endpoint to upgrade even if it will not take advantage of Version 6 features to practice the upgrade process. 
+
+
+### Move to .NET 4.5.2
 
 The minimum .NET version for NServiceBus Version 6 is .NET 4.5.2.
 
-**Users must update all projects (that reference NServiceBus) to .NET 4.5.2 before updating to NServiceBus Version 6.**
+**All projects (that reference NServiceBus) must be updated to .NET 4.5.2 before updating to NServiceBus Version 6.**
```

Comment:
should we make use of this https://blogs.msdn.microsoft.com/dotnet/2014/08/07/moving-to-the-net-framework-4-5-2/  ?